subsys/bluetooth/shell/bt.c

```diff
@@ -1653,25 +1653,25 @@ static int cmd_per_adv_sync_delete(const struct shell *shell, size_t argc,
 {
 	int err;
 	int index = 0;
-	struct bt_le_per_adv_sync **per_adv_sync = NULL;
+	struct bt_le_per_adv_sync *per_adv_sync = NULL;
 
 	if (argc > 1) {
 		index = strtol(argv[1], NULL, 10);
 	}
 
-	per_adv_sync = &per_adv_syncs[index];
+	per_adv_sync = per_adv_syncs[index];
 
 	if (!per_adv_sync) {
 		return -EINVAL;
 	}
```

Comment:
I know it's not directly related to the specific thing you're trying to fix, but isn't there a missing check for `index < ARRAY_SIZE(per_adv_syncs)` here?

Reply:
Absolutely! Thanks for finding that.